include/picongpu/fields/currentDeposition/Esirkepov/Esirkepov.hpp

```diff
@@ -167,15 +198,15 @@ namespace picongpu
                  *     ( this helps the compiler to mask threads without work )
                  */
                 for(int i = begin; i < end + 1; ++i)
-                    if(i < end + leaveCell[0])
+                    if(i < end + bitpacking::getValue(parStatus[0], bitpacking::Status::LEAVE_CELL))
```

Comment:
Does it make sense to save `bitpacking::getValue(parStatus[0], bitpacking::Status::LEAVE_CELL)` to a variable since it does not change? (then could also be used earlier to construct variables like  `shapeI`)